File: controller function.m
- (void)userContentController:(WKUserContentController *)userContentController
      didReceiveScriptMessage:(WKScriptMessage *)message {
  if ([message.body[@"command"] isEqual:@"setUserProperty"]) {
      for (id key in message.body[@"parameters"]){
          if([key isEqual:@"user_id"]){
              printf("user_id   : %s \n",[[message.body[@"parameters"] objectForKey:key] UTF8String] );
              [FIRAnalytics setUserID:[message.body[@"parameters"] objectForKey:key]];
          }
          else{
             
              [FIRAnalytics setUserPropertyString:[message.body[@"parameters"] objectForKey:key] forName:key];
          }
      }
      
  } else if ([message.body[@"command"] isEqual: @"logEvent"]) {
      [FIRAnalytics logEventWithName:message.body[@"name"] parameters:message.body[@"parameters"]];

  }
}

File: controller function.swift
func userContentController(_ userContentController: WKUserContentController,
                                 didReceive message: WKScriptMessage) {
          guard let body = message.body as? [String: Any] else { return }
          guard let command = body["command"] as? String else { return }
          
        
          if command == "setUserProperty" {
            guard let params = body["parameters"] as? [String: NSObject] else { return }
            
            for (key, value) in params {
                if(key == "user_id"){
                    Analytics.setUserID(value as? String)
                    
                } else{
                    Analytics.setUserProperty(value as? String, forName: key)
                }
            }
          } else if command == "logEvent" {
              guard let name = body["name"] as? String else { return }
              guard let params = body["parameters"] as? [String: NSObject] else { return }
            Analytics.logEvent(name, parameters: params)
          }
        }
